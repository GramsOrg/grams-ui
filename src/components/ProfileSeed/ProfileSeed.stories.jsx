import React from 'react';

import { action } from "@storybook/addon-actions";

import ProfileSeed from './ProfileSeed.tsx';

export default {
  title: 'Onboarding/ProfileSeed',
  component: ProfileSeed,
  parameters: {
    componentSubtitle: 'Displays the 24-word seed to the user and prompts to accept securing the seed',
    storyshots: { disable: false },
  },
  argTypes: {
    dir: {
      control: 'radio',
      options: ['ltr', 'rtl']
    },
    inverted: {
      control: 'boolean'
    }
  }
};

const seed = 'hello world again potato tomato whisper galal cloud profile oscar eleven twelve thirteen amazing great hungry available option bamya lorem ipsum dictionary developer snapshot';

const Template = ({dir, inverted, ...props}) => {
  return (
    <ProfileSeed
      dir={dir}
      inverted={inverted}
      seed={seed}
      {...props}
    />
  );
};

export const Default = Template.bind({});
Default.args = {
  dir: 'ltr',
  inverted: false
}

export const RTL = Template.bind({});
RTL.args = {
  dir: 'rtl',
  inverted: false
}
