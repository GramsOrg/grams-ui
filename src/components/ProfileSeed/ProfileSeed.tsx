import React, { useState } from 'react';

import { Button, Checkbox, Grid, Segment } from 'semantic-ui-react';

import './ProfileSeed.css';

export interface IProfileSeedProps {

  /**
   * Defines the direction of the component
   */
  dir?: 'ltr' | 'rtl';

  /**
   * Toggles between light and dark modes
   */
  inverted?: boolean;

  /**
   * The 24-word seed string to display to the user.
   */
  seed: string;

  /**
   * A callback function to be called when the user confirms that they have
   * saved the seed.
   */
  onConfirm?: () => void;
}

const defaultProps = {
  dir: 'ltr',
  inverted: false
};

const ProfileSeed = (props: IProfileSeedProps) => {

  const seedWords = props.seed.split(' ');
  const [confirm, setConfirm] = useState(false);

  return (
    <Segment basic>
      <Grid columns={3} centered>
        {seedWords.map((word, index) => (
          <Grid.Column key={index}>
            <div>{word}</div>
          </Grid.Column>
        ))}
      </Grid>
      <Checkbox
        label='I understand that Grams will not be able to show me this seed again and that I have saved it in a secure place'
        checked={confirm}
        onChange={(e, { checked }) => setConfirm(!!checked)} 
      />
      <Button
        primary
        fluid
        onClick={props.onConfirm}
        disabled={!confirm}
      >
        Continue
      </Button>
    </Segment>
  );
};

ProfileSeed.defaultProps = defaultProps;

export default ProfileSeed;
