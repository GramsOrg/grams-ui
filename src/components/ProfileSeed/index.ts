import ProfileSeed from './ProfileSeed';
export default ProfileSeed;
