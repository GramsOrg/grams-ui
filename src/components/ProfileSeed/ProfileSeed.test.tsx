/**
 * @jest-environment jsdom
 */

import React from 'react';
import { describe, expect, test } from '@jest/globals';
import { render } from '@testing-library/react';

import ProfileSeed from './ProfileSeed';

describe('ProfileSeed Component', () => {
  test('it should match the snapshot', () => {
    const { asFragment } = render(<ProfileSeed seed='any' />);
    expect(asFragment()).toMatchSnapshot();
  });
});
