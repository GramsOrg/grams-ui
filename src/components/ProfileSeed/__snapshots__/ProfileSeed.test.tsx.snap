// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileSeed Component it should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="ui basic segment"
  >
    <div
      class="ui centered three column grid"
    >
      <div
        class="column"
      >
        <div>
          any
        </div>
      </div>
    </div>
    <div
      class="ui checkbox"
    >
      <input
        class="hidden"
        readonly=""
        tabindex="0"
        type="checkbox"
        value=""
      />
      <label>
        I understand that Grams will not be able to show me this seed again and that I have saved it in a secure place
      </label>
    </div>
    <button
      class="ui fluid primary disabled button"
      disabled=""
      tabindex="-1"
    >
      Continue
    </button>
  </div>
</DocumentFragment>
`;
